In [1]:
# import libraries
import pandas as pd
import numpy as np

In [2]:
# load data
heart = pd.read_csv('heart_disease.csv')
yes_hd = heart[heart.heart_disease == 'presence']
no_hd = heart[heart.heart_disease == 'absence']

---

### Exercise 01
In general, total cholesterol over 240 mg/dl is considered “high” (and therefore unhealthy). Calculate the mean cholesterol level for patients who were diagnosed with heart disease and print it out. Is it higher than 240 mg/dl?

In [3]:
mean_chol_yes_hd = yes_hd['chol'].mean()
print(round(mean_chol_yes_hd, 2))


251.47


The mean cholesterol level for patients diagnosed with heart disease is 251.47 mg/dl, which is higher than 240 mg/dl. Therefore, on average, these patients have high cholesterol.

---
### Exercise 02
Do people with heart disease have high cholesterol levels (greater than or equal to 240 mg/dl) on average? Import the function from scipy.stats that you can use to test the following null and alternative hypotheses:

- Null: People with heart disease have an average cholesterol level equal to 240 mg/dl
- Alternative: People with heart disease have an average cholesterol level that is greater than 240 mg/dl

In [4]:
from scipy import stats

cholesterol_values_yes_hd = yes_hd['chol'].dropna()


t_stat, p_value = stats.ttest_1samp(cholesterol_values_yes_hd, popmean=240, alternative='greater')

print(f"p-value: {p_value}")


p-value: 0.0035411033905155707


Since the p-value is well below 0.05, you can reject the null hypothesis.

Conclusion: People with heart disease in your dataset have an average cholesterol level significantly greater than 240 mg/dl, meaning they tend to have high cholesterol.

---

### Exercise 03

In general, total cholesterol over 240 mg/dl is considered “high” (and therefore unhealthy). Calculate the mean cholesterol level for patients who were **not** diagnosed with heart disease and print it out. Is it higher than 240 mg/dl?

In [5]:
mean_chol_no_hd = no_hd['chol'].mean()
print(round(mean_chol_no_hd, 2))

242.64


The mean cholesterol level for patients diagnosed **without** heart disease is 242.64 mg/dl, which is slightly higher than 240 mg/dl. Therefore, on average, these patients do not have evidence of high cholesterol relative to the 240 mg/dl threshold.

---

### Exercise 04

Do people **without** heart disease have high cholesterol levels (greater than or equal to 240 mg/dl) on average? Import the function from scipy.stats that you can use to test the following null and alternative hypotheses:

- Null: People **without** heart disease have an average cholesterol level equal to 240 mg/dl
- Alternative: People **without** heart disease have an average cholesterol level that is greater than 240 mg/dl

In [6]:
from scipy import stats

cholesterol_values_no_hd = no_hd['chol'].dropna()

t_stat, p_value = stats.ttest_1samp(cholesterol_values_no_hd, popmean=240, alternative='greater')

print(f"p-value: {p_value}")


p-value: 0.26397120232220506


Since the p-value is well above 0.05, you cannot reject the null hypothesis.

Conclusion: People without heart disease in your dataset have an average cholesterol level not significantly greater than 240 mg/dl. In other words, although the mean is slightly above the threshold, statistically it is consistent with the 240 mg/dl threshold.

---

### Exercise 05


Let’s now return to the full dataset (saved as heart). How many patients are there in this dataset? Save the number of patients as num_patients and print it out.


In [16]:
num_patients = len(heart)
print(f"There is a total of {num_patients} patients")

There is a total of 303 patients


---
### Exercise 06

Remember that the fbs column of this dataset indicates whether or not a patient’s fasting blood sugar was greater than 120 mg/dl (1 means that their fasting blood sugar was greater than 120 mg/dl; 0 means it was less than or equal to 120 mg/dl).

Calculate the number of patients with fasting blood sugar greater than 120. Save this number as num_highfbs_patients and print it out.

In [25]:
num_highfbs_patients = round(np.sum(heart.fbs))
print(f"There are {num_highfbs_patients} patients with fasting blood sugar greater than 120")

There are 45 patients with fasting blood sugar greater than 120


---

### Exercise 08

Sometimes, part of an analysis will involve comparing a sample to known population values to see if the sample appears to be representative of the general population.

By some estimates, about 8% of the U.S. population had diabetes (diagnosed or undiagnosed) in 1988 when this data was collected. While there are multiple tests that contribute to a diabetes diagnosis, fasting blood sugar levels greater than 120 mg/dl can be indicative of diabetes (or at least, pre-diabetes). If this sample were representative of the population, approximately how many people would you expect to have diabetes? Calculate and print out this number.

Is this value similar to the number of patients with a resting blood sugar above 120 mg/dl — or different?

In [32]:
expected_diabetes = 0.08 * num_patients

print(f"Expected number of patients with diabetes (8%): {round(expected_diabetes)}")

Expected number of patients with diabetes (8%): 24


The observed number is almost double the expected number, so the sample does not match what you’d expect if it were perfectly representative of the general population.

---

### Exercise 09

Does this sample come from a population in which the rate of fbs > 120 mg/dl is equal to 8%? Import the function from scipy.stats that you can use to test the following null and alternative hypotheses:

Null: This sample was drawn from a population where 8% of people have fasting blood sugar > 120 mg/dl
Alternative: This sample was drawn from a population where more than 8% of people have fasting blood sugar > 120 mg/dl

In [35]:
from scipy.stats import binomtest

---

### Exercise 10

Run the hypothesis test indicated in task 9 and print out the p-value. Using a significance threshold of 0.05, can you conclude that this sample was drawn from a population where the rate of fasting blood sugar > 120 mg/dl is significantly greater than 8%?

In [63]:
pval = binomtest(num_highfbs_patients, num_patients, .08, alternative='greater').pvalue

print(f"{pval:.6f}")

0.000047


The p-value is far smaller than 0.05,the null hypothesis is rejected. The sample provides strong evidence that the proportion of patients with fasting blood sugar > 120 mg/dl is significantly higher than 8%, and the sample does not appear to come from a population where only 8% have high FBS.